lonboard/traits.py

```diff
@@ -705,3 +706,99 @@ def validate(self, obj, value) -> Union[float, pa.ChunkedArray, pa.DoubleArray]:
 
         self.error(obj, value)
         assert False
+
+
+class NormalAccessor(FixedErrorTraitType):
+    """
+    A representation of a deck.gl normal accessor
+
+    Acceptable inputs:
+    - A numpy ndarray with two dimensions: The size of the
+    second dimension must be 3 i.e. `(N,3)`
+    - a pyarrow `FixedSizeListArray` or
+    `ChunkedArray` containing `FixedSizeListArray`s
+    where the size of the inner fixed size list 3.
+    """
+
+    default_value = [0, 0, 1]
+    info_text = (
+        "List representing normal of each object, in [nx, ny, nz]. or numpy ndarray or "
+        "pyarrow FixedSizeList representing the normal of each object, in [nx, ny, nz]"
+    )
+
+    def __init__(
+        self: TraitType,
+        *args,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(*args, **kwargs)
+        self.tag(sync=True, **ACCESSOR_SERIALIZATION)
+
+    def validate(
+        self, obj, value
+    ) -> Union[Tuple[int, ...], List[int], pa.ChunkedArray, pa.FixedSizeListArray]:
+        """
+        Values in acceptable types must be contiguous
+        (the same length for all values)
+        """
+
+        fixed_list_size = 3
+
+        if isinstance(value, List):
+            if len(value) != 3:
+                self.error("Normal array must have length 3, (x,y,z)")
+
+            if not all(isinstance(item, (int, float)) for item in value):
+                self.error("All elements of Normal array must be int or float type")
+
+            return pa.FixedSizeListArray.from_arrays(value, fixed_list_size)
+
+        if isinstance(value, np.ndarray):
+            if value.ndim != 2 or value.shape[1] != 3:
+                self.error(obj, value, info="Normal array must be 2D with shape (N,3)")
+
+            if not np.any(value == 1):
+                self.error(
+                    obj, value, info="One of the normal vector elements must be 1."
+                )
+
+            if not np.issubdtype(value.dtype, np.float32):
+                warnings.warn(
+                    """Warning: Numpy array should be floating point type.
+                    Converting to float32 point pyarrow array"""
+                )
+                value = value.astype(np.float(pa.list_(pa.float32(), 3)))
```

Comment:
This will fail; you want `value.astype(np.float32)`. You can't pass a pyarrow type into a numpy function like this.